app/src/main/java/com/tarmeez/game/auth/RegisterFragment.kt

```diff
@@ -0,0 +1,113 @@
+package com.tarmeez.game.auth
+
+import android.app.AlertDialog
+import android.os.Bundle
+import android.util.Log
+import android.util.Patterns
+import android.view.*
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.Navigation
+import com.tarmeez.game.R
+import com.tarmeez.game.databinding.FragmentRegisterBinding
+import dagger.hilt.android.AndroidEntryPoint
+
+private const val TAG = "RegisterFragment"
+@AndroidEntryPoint
+class RegisterFragment: Fragment() {
+    private var _binding:FragmentRegisterBinding? = null
+    private val binding:FragmentRegisterBinding?
+    get() = _binding!!
+    private val authViewModel: AuthViewModel by viewModels()
+    private lateinit var progressDialog: AlertDialog
+
+    override fun onCreateView (
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentRegisterBinding.inflate(inflater, container, false)
+        initViews()
+        initObserver()
+        return binding?.root
+    }
+
+    private fun initViews() {
+        binding?.userEmail?.hint = getString(R.string.enter_email, " ")
```

Comment:
There is any necessary reason to put the hint programmatically? .. if no please adding hint from layout to make code more clear

Reply:
yes, the drawable icon was very close to the hint, so the space has to be set programmatically.